.github/workflows/ci-nx-dev-e2e.yml

```diff
@@ -0,0 +1,71 @@
+name: CI nx-dev e2e
+
+on: pull_request
+
+jobs:
+  determine_if_nx_dev_will_be_deployed:
+    runs-on: ubuntu-latest
+    name: Determine if an nx-dev preview deployment will occur
+    outputs:
+      will_be_deployed: ${{ steps.check_will_be_deployed.outputs.will_be_deployed }}
+    steps:
+      - uses: actions/checkout@v3
+        name: Checkout
+        with:
+          # By default, PRs will be checked-out based on the Merge Commit, but we want the actual branch HEAD.
+          ref: ${{ github.event.pull_request.head.sha }}
+          # We need to fetch all branches and commits so that Nx affected has a base to compare against.
+          fetch-depth: 0
+
+      - name: Determine if nx-dev deployment will occur based on the result of the same script we use in Vercel
+        id: check_will_be_deployed
+        run: |
+          WILL_BE_DEPLOYED=$(./scripts/vercel/ignore-build.sh nx-dev && echo "" || echo "true")
+          echo "::set-output name=will_be_deployed::$WILL_BE_DEPLOYED"
+        shell: bash
+
+  nx_dev_e2e_tests:
```

Comment:
Can we use the `file-server` to serve the outputs of `nx build nx-dev` and test with that server? Let's not point to the deployed version.

Reply:
https://github.com/nrwl/nx/pull/10088